src/Infrastructure/ErrorLoggingFileFetcher.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\Fundraising\Frontend\Infrastructure;
+
+use FileFetcher\FileFetcher;
+use FileFetcher\FileFetchingException;
+use Psr\Log\LoggerAwareInterface;
+use Psr\Log\LoggerInterface;
+
+class ErrorLoggingFileFetcher implements FileFetcher, LoggerAwareInterface {
+
+	private $wrappedFileFetcher;
+
+	private $logger;
+
+	public function __construct( FileFetcher $fileFetcher, LoggerInterface $logger ) {
+		$this->wrappedFileFetcher = $fileFetcher;
+		$this->logger = $logger;
+	}
+
+	public function fetchFile( $fileUrl ) {
+		try {
+			return $this->wrappedFileFetcher->fetchFile( $fileUrl );
+		} catch ( FileFetchingException $e ) {
+			$this->logger->error( $e->getMessage(), [
+				'code' => $e->getCode(),
+				'trace' => $e->getTraceAsString()
+			] );
+			return '';
```

Comment:
> I could do a try...catch in the WordListFileReader instead

What about doing one there "as well" instead of "instead"? You would not need to log there, just catch the error, and have that code decide that the error should not bring down the app. Isn't that a better place to have this decision made? That also leaves you with a decorator that does not have an And in its name.